File: qqbot-gateway/src/main/java/com/mieai/qqbot/gateway/GatewaySessionException.java
package com.mieai.qqbot.gateway;

import java.util.Objects;

/** Sanitized session-layer failure with a stable reconnect category. */
public final class GatewaySessionException extends RuntimeException {
    private final GatewayReconnectCause category;

    public GatewaySessionException(
            GatewayReconnectCause category, String message, Throwable cause) {
        super(message, cause);
        this.category = Objects.requireNonNull(category, "category must not be null");
    }

    public GatewayReconnectCause category() {
        return category;
    }
}

File: qqbot-gateway/src/main/kotlin/com/mieai/qqbot/gateway/GatewaySessionException.kt
package com.mieai.qqbot.gateway

/** Sanitized session-layer failure with a stable reconnect category. */
class GatewaySessionException(
    category: GatewayReconnectCause,
    message: String?,
    cause: Throwable?,
) : RuntimeException(message, cause) {
    private val categoryValue = category

    fun category(): GatewayReconnectCause = categoryValue
}
